### ISSUE TITLE
K:\ drive becomes unavailable after a minute of starting Keybase.

### ISSUE THREAD

**Issue by martinmicevski** [keybase/client#24388]
[write something useful and descriptive here]

my log id: d42146925ff4c792ed29681c

**Comment by martinmicevski** [comment]
Hello there,

I fix this issue.
I did complete uninstall (with RevoUninstaller) and I installed again. Now seem that everything is working right.

Martin

**Comment by GitGoppy** [comment]
Hey, I got the same problem, appeared only yesterday out of nowhere, uninstalled app completely using revo and tried reinstalling dokan, also did restarts between each action to make sure nothing was left in memory and driver was properly uninstalled. Did this multiple times in multiple ways and kbfsmount stops working after a minute. Exiting and starting the client makes it work for one minute after which it get's disabled and I have no idea why. Tried looking in kbfslog but not making anything clear. Did you do anything else to fix your problem?

**Comment by martinmicevski** [comment]
> Hey, I got the same problem, appeared only yesterday out of nowhere, uninstalled app completely using revo and tried reinstalling dokan, also did restarts between each action to make sure nothing was left in memory and driver was properly uninstalled. Did this multiple times in multiple ways and kbfsmount stops working after a minute. Exiting and starting the client makes it work for one minute after which it get's disabled and I have no idea why. Tried looking in kbfslog but not making anything clear. Did you do anything else to fix your problem?

Hi there,

Can you try to go Settings -> Apps & Features -> find Keybase and try Modify. This should fix the problem temp.

What version of Dokan did you install?

BR,
Martin

**Comment by GitGoppy** [comment]
> > Hey, I got the same problem, appeared only yesterday out of nowhere, uninstalled app completely using revo and tried reinstalling dokan, also did restarts between each action to make sure nothing was left in memory and driver was properly uninstalled. Did this multiple times in multiple ways and kbfsmount stops working after a minute. Exiting and starting the client makes it work for one minute after which it get's disabled and I have no idea why. Tried looking in kbfslog but not making anything clear. Did you do anything else to fix your problem?
> 
> Hi there,
> 
> Can you try to go Settings -> Apps & Features -> find Keybase and try Modify. This should fix the problem temp.
> 
> What version of Dokan did you install?
> 
> BR,
> Martin

Yes, the modify option fixes the issue, and quitting and starting the app also fixes it. But the fix only works for one minute after which the drive becomes unavailable again. 

I currently use dokan version 1.2.2.1000 as made available by keybase from settings
I tried using different versions of dokan from dokany github repo, I tried the newest version of dokan 1.4.0.1000 and version 1.2.1.2000 which was the version before the current one.
Every version has the exact same behaviour of working for one minute and then becoming inaccessible.
I tried killing kbfsdokan process because keybase then creates a new kbfsdokan process, but that only fixes it for one minute again.
I tried using kbfsdokan command and then setting a different mountpoint, this also works only for one minute.
also when using kbfsdokan command in cmd prompt you get an active log where any errors by kbfsdokan are visible,
KBFSdokan reports SID match errors before reauthenticating and then changing it's mount point after which there are no errors even after the drive becomes inaccesible.
keybase kbfsmount status gives this output
`- INFO detectDokanDll: returning true
{
  "version": "",
  "bundleVersion": "",
  "kextID": "",
  "path": "",
  "kextStarted": true,
  "installStatus": 4,
  "installAction": 1,
  "mountInfos": null,
  "status": {
    "code": 0,
    "name": "",
    "desc": "",
    "fields": [
      {
        "key": "uninstallString",
        "value": "\"C:\\ProgramData\\Package Cache\\{926ca734-e343-434b-904d-e8b372e32c97}\\DokanSetup.exe\"  /uninstall"
      }
    ]
  }
}`
When I installed versions of dokan from the official dokan repo however I noticed that it spammed my event log.
There were hundreds of events made by dokan1 that had only "Mounting disk device" message in description. This does not happen in keybase's version of dokan probably because some kind of debug is turned off.
I also found the event where disk was succesfully mounted and about a minute later an event that said that the disk was unmounted. There was no information as to what caused the unmount. 
The "Mounting disk device" spam was also happening during the one minute period between mount and unmount while the drive was working so unsure if that is the reason.

Keybase client normally sees files in files section and command "keybase kbfsmount get" gives the correct drive letter for the mount point. So I presume the very little amount of SID errors are not the culprit, since SID errors would prevent me from seeing and making the files through the client.

I have installed new audio and grpahics drivers yesterday (Nvidia and Realtek), no other programs were installed or updated.
Windows did not receive any updates before the problem started.

Tried disabling antivirus, no effect, antivirus did not log any interference with dokan or keybase. Not the culprit

I have absolutely no idea how or why this started happening. I require advice.

**Comment by GitGoppy** [comment]
I did command kbfsdokan with mountflags set to 3 and debug. When the thing crashes it sends this message. 
`Ioctl failed for wait with code 995` and then thread terminated message.
I think it has to do with some kind of keepalive timeout that dokan uses and the timeout interval can be somehow extended I think which would fix the issue.
So if anyone knows how to extend the IOCTL timeout do tell.

**Comment by GitGoppy** [comment]
**CONFIRMED ORIGIN**
If we go to the dokan install directory in "C:\Program Files\Dokan\Dokan Library-1.2.2\"
we can execute program dokanctl.exe in command prompt to get info.
`dokanctl.exe /l a` command will give us a list of current mountpoints mounted by Dokan

The K drive is active for about a minute during which `dokanctl.exe /l a` shows that the drive is mounted
When the `Ioctl failed for wait with code 995` error occurs the drive stops working in file explorer and `dokanctl.exe /l a`  shows no mounted drives.

I don't know what's causing the Ioctl timeout but only hope is to extend the time required for timeout and the drive would have enough time to do whatever operation it has to do and that would be a potential fix. However I don't know how to extend this timeout so if anyone could help it would be nice.

**Comment by martinmicevski** [comment]
> **CONFIRMED ORIGIN**
> If we go to the dokan install directory in "C:\Program Files\Dokan\Dokan Library-1.2.2"
> we can execute program dokanctl.exe in command prompt to get info.
> `dokanctl.exe /l a` command will give us a list of current mountpoints mounted by Dokan
> 
> The K drive is active for about a minute during which `dokanctl.exe /l a` shows that the drive is mounted
> When the `Ioctl failed for wait with code 995` error occurs the drive stops working in file explorer and `dokanctl.exe /l a` shows no mounted drives.
> 
> I don't know what's causing the Ioctl timeout but only hope is to extend the time required for timeout and the drive would have enough time to do whatever operation it has to do and that would be a potential fix. However I don't know how to extend this timeout so if anyone could help it would be nice.

Did you try to remove Keybase and Dokan with RevoUninstaller and after that just to install only the new version of Keybase?

**Comment by GitGoppy** [comment]
Yes, the newest one from keybase website for windows. Uninstalled completely with revouninstaller  both dokan and keybase and used advanced scan to remove any registry or file traces. Restarted PC, installed the newest version and installed Dokan from keybase settings. Still the same behaviour.

**Comment by GitGoppy** [comment]
What???? HOW WHY???? I quit the broken keybase app and tried running it in sandboxie (a program sandbox with which you can test safety of software and stuff), my profile for the sandbox is so that programs can access storage normally but new stuff is all stored inside the sandbox storage but their process is a part of the sandbox I guess (??? IDK how this works) and the K: drive didn't crash when keybase was run inside sandboxie. I have ABSOLUTELY no IDEA WHY. I could guess that there is an incompatible driver or something like that. But I need to know what is the root cause.

**Comment by JPLeoRX** [comment]
I'm experiencing the same behavior, disk K just keeps crashing on me. This effectively makes the cloud storage part of Keybase unusable on Windows machines.
